src/ui/src/builder/BuilderSettingsHandlers.vue

```diff
@@ -186,7 +186,13 @@ const recognisedEvents: ComputedRef<WriterComponentDefinition["events"]> =
 	});
 
 const userFunctions = computed(() => wf.getUserFunctions());
-const pageKeys = computed(() => wf.getPageKeys());
+const pageKeys = computed(() => {
+	const pages = wf.getComponents("root");
+	const pageKeys = pages
+		.map((page) => page.content["key"])
+		.filter((pageKey) => !!pageKey);
```

Comment:
I don't see how this would work

Reply:
`Boolean` is a function, so it can be passed as a callback. Basically, it is equivalent to your code. 

```js
[0,false, '', null].filter((v) => !!v);
[0,false, '', null].filter((v) => Boolean(v));
[0,false, '', null].filter(Boolean);
```

But if you don't like the syntax, you can omit this comment.

Reply:
You're right, took me a while. I'm too used to seeing an arrow function. 

```js
[0,false, '', null].filter((v) => Boolean(v));
```

is probably the best one/most expressive.